src/What4/Serialize/SETokens.hs

```diff
@@ -205,9 +220,40 @@ parseBV = P.char '#' >> ((P.char 'b' >> parseBin) P.<|> (P.char 'x' >> parseHex)
 
         parseHex = (\s -> (length s * 4, read ("0x" ++ s))) <$> P.many1 P.hexDigit
 
+parseFloat :: Parser (Some W4.FloatPrecisionRepr, BF.BigFloat)
+parseFloat = do
+  _ <- P.string "#f#"
+  -- We printed the nat reprs out in decimal
+  eb :: Natural
+     <- read <$> P.many1 P.digit
+  _ <- P.char '#'
+  sb :: Natural
+     <- read <$> P.many1 P.digit
+  _ <- P.char '#'
+
+  -- The float value itself is printed out as a hex literal
+  hexDigits <- P.many1 P.hexDigit
+
+  Some ebRepr <- return (PN.mkNatRepr eb)
+  Some sbRepr <- return (PN.mkNatRepr sb)
+  case (PN.testLeq (PN.knownNat @2) ebRepr, PN.testLeq (PN.knownNat @2) sbRepr) of
+    (Just PN.LeqProof, Just PN.LeqProof) -> do
+      let rep = W4.FloatingPointPrecisionRepr ebRepr sbRepr
+
+      -- We know our format: it is determined by the exponent bits (eb) and the
+      -- significand bits (sb) parsed above
+      let fmt = BF.precBits (fromIntegral sb) <> BF.expBits (fromIntegral eb)
+      let (bf, status) = BF.bfFromString 16 fmt hexDigits
+      case status of
+        BF.Ok -> return (Some rep, bf)
+        _ -> P.unexpected ("Error parsing hex float: 0x" ++ hexDigits)
+    _ -> P.unexpected ("Invalid exponent or significand size: " ++ show (eb, sb))
+
+
```

Comment:
I'm saddened to see this duplicated between here and the FastSExpr.hs file.  Is there no way to unify these?

Reply:
At this point we don't actually use this one anymore - we could just delete it. We can't really unify them because the parsing functions are not at all related, even though the names are similar, between parsec and megaparsec.